### PR TITLE
Add serve command

### PR DESCRIPTION
**Pull request by spiegela**
Add new `serve` command to launch actix web based webserver supporting juniper graphql server. Also adds `TypeKind` field for `DataType` so that Manutara knows how to handle certain types, and adds a stub for dynamically generated schema based on `DataType` resources.

### CODE REVIEW

File: libs/schema/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    pub version: String,
+}
+
+pub fn schema(_project_name: String, _version: String) -> Schema {
+    Schema::new(
+        Query,
+        EmptyMutation::<Context>::new(),
+        EmptySubscription::<Context>::new(),
+    )
+}
```
Comment (Bjohnson131) [inline libs/schema/src/lib.rs:17]: an (almost) empty schema?

File: libs/server/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            .service(graphql)
+            .service(graphql_playground)
+            // the graphiql UI requires CORS to be enabled
+            .wrap(Cors::permissive())
+            .wrap(middleware::Logger::default())
+    })
+    .workers(workers)
+    .bind((address, port))?
+    .run()
+    .await
```
Comment (Bjohnson131) [inline libs/server/src/lib.rs:54]: this blocks until the server ends? is this a property of whatever run() returns?

File: libs/server/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+) -> io::Result<()> {
+    // Create Juniper schema
+    let schema = Arc::new(schema::schema(project_name, version));
+
+    log::info!("starting HTTP server on port {}", port);
+    log::info!("GraphQL API: http://{}:{}", address, port);
+    log::info!("GraphiQL playground: http://{}:{}/graphiql", address, port);
+
+    // Start HTTP server
+    HttpServer::new(move || {
```
Comment (Bjohnson131) [inline libs/server/src/lib.rs:42]: why move a struct literal? Stack Exchange says that the move keyword removes the lifetime expectancy and replaces it with 'static, as well as changing the owner to whatever you're passing the data into... Why is that needed if we're blocking until the thread is done anyway?

**Review by Bjohnson131 — APPROVED** [review]

File: libs/schema/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "schema"
+version = "0.1.0"
+edition = "2021"
```
[the remark this answers is not in the document]
Comment (spiegela) [reply libs/schema/Cargo.toml:4]: It's the version of the `Cargo.toml` schema

File: libs/schema/src/lib.rs (hunk 1 above)
Comment (spiegela) [reply libs/schema/src/lib.rs:17]: Yeah, I just included a current version query. I don't think there's a `juniper::EmptyQuery` object, since it would be an invalid schema to have no queries

File: libs/sdl/src/schema.rs [hunk 5]
```diff
@@ -202,16 +218,32 @@ impl DataType {
...
-                _ => {}
+        handle_directives(&obj.directives, &mut d);
+        d
+    }
+}
+
+fn handle_directives(directives: &[Directive<String>], data_type: &mut DataType) {
+    for directive in directives {
+        match directive {
+            Directive {
```
[the remark this answers is not in the document]
Comment (spiegela) [reply libs/sdl/src/schema.rs:229]: Yep. That's a good suggestion, I'll do that.

File: libs/server/src/lib.rs (hunk 2 above)
Comment (spiegela) [reply libs/server/src/lib.rs:54]: It should not return unless the server fails since this is the webserver thread -- it'll run until killed unless something bad happens.

File: libs/server/src/lib.rs (hunk 3 above)
Comment (spiegela) [reply libs/server/src/lib.rs:42]: Yeah, I was following the [juniper actix web](https://github.com/actix/examples/blob/master/graphql/juniper/src/main.rs) example. The `schema` variable doesn't outlive the http server without moving ownership.

File: libs/binding/src/lib.rs [hunk 6]
```diff
@@ -70,7 +70,7 @@ impl<'a> Project<'a> {
     pub fn generate(&self, data_types: &[DataType]) -> Result<()> {
         for lang in &self.languages {
             let constructor = GENERATORS.get(lang.to_string().as_str()).copied().unwrap();
-            let generator = constructor(self.name, self.version);
+            let generator = constructor(self.name.as_str(), self.version);
```
[the remark this answers is not in the document]
Comment (spiegela) [reply libs/binding/src/lib.rs:73]: Because the resulting `BindingGenerator` can be a consistent size if it's passed a ref instead of a full string
